### DISCUSSION TITLE
imagemagick png losing alpha chanel on save

### DISCUSSION THREAD

**Discussion by owenaled** [ImageMagick/ImageMagick#6906] [Help] [answered]
### ImageMagick version

7.1.1-21

### Operating system, version and so on

MaOS 13.5.2 

### Description

I have a greyscale jpeg of a line drawing I need to convert to a white with transparency and save as png
I have been using this for a while and was working perfectly

`magick input.jpeg -level 40%,75%,0.8 -negate -alpha copy -fill white -colorize 100 output.png`

but at some point it has stopped working and just produces a solid white opaque image

if I change the output to tiff it works as intended
`magick input.jpeg -level 40%,75%,0.8 -negate -alpha copy -fill white -colorize 100 output.tiff`

I can convert the tiff to png and it works ok, but I would prefer to skip that step if possible

Thanks in advance for any help

input
![input](https://github.com/ImageMagick/ImageMagick/assets/36334585/92847b48-0461-4293-8615-0ec34882d7b5)

tiff output ( checkerboad added to visuallise)
![output-correct](https://github.com/ImageMagick/ImageMagick/assets/36334585/49b959ed-4e8f-4752-ad79-d8796c36ccef)

**Comment by snibgo** [comment, marked as answer]
I confirm the problem in IM v7.1.1-22 on Windows 11.

This seems to be a bug in `-alpha copy`, not setting some metadata that the PNG encoder wants.

A workaround is to insert `-alpha activate` after `-alpha copy`. Here is a simple example:

```
C:\web\im>%IMG7%magick xc:red -alpha copy x.png

C:\web\im>%IMG7%magick x.png txt:
# ImageMagick pixel enumeration: 1,1,0,255,srgb
0,0: (255,0,0)  #FF0000  red  <=== This is wrong

C:\web\im>%IMG7%magick xc:red -alpha copy -alpha activate x.png

C:\web\im>%IMG7%magick x.png txt:
# ImageMagick pixel enumeration: 1,1,0,255,srgba
0,0: (255,0,0,54)  #FF000036  srgba(255,0,0,0.211765)  <=== This is correct
```

**Reply by coeing** [reply]
Is there a fix already or an issue? I got this bug in version 7.1.1-32 on Unix.

**Reply by fmw42** [reply]
I do not know, but first thing to do is to check the changeling at https://github.com/ImageMagick/Website/blob/main/ChangeLog.md

**Comment by owenaled** [comment]
Yes, that works, Thank you so much
